### PR TITLE
Rewrote form widget tests as proper unittests.

### PR DESCRIPTION
**Pull request by prestontimmons**

### CODE REVIEW

**Comment by prestontimmons** [comment]
Here's an overview of where tests are moved to:

``FormsWidgetTests``
* Tests are broken up and moved to individual files in ``widget_tests``.
* Duplicate tests removed from widgets that inherit ``TextInput`` and ``Select``. For example, we don't need to test contstructor attrs handling in every widget, or that generators work as a ``choices`` argument in every ``Select`` based widget. It's enough to do this once in the parent widget.
* ``test_multi`` and ``test_multiwidget`` are split into ``widget_tests/test_multiwidget.py`` and ``field_tests/test_multivaluefield.py``. The widget and field tests were previously mingled together.

``FormsI18NWidgetsTests``
* These are moved into the individual widget test files with a name similar to ``test_l10n``.

``WidgetTests``
* ``test_12048`` moved to ``widget_tests.test_multiwidget.test_deepcopy``

``ClearableFileInputTests``
* Moved into ``test_clearablefileinput.py`` except for one case: ``test_render_custom_template``.

``SelectDateWidgetTests``
* Split into ``widget_tests/test_selectdatewidget.py`` and ``field_tests.test_datefield.py``. The widget and field tests were previously mingled together.

``SelectWidgetTests``
* Moved to ``widget_tests/test_select.py``

The following tests were kept in the ``forms_tests/tests/test_widget.py`` file:

``FormsWidgetTests.test_radiofieldrenderer``
``FormsWidgetTests.test_subwidget``
``FormsWidgetTests.test_sub_widget_html_safe``
``FormsWidgetTests.test_choice_input_html_safe``
``FormsWidgetTests.test_choice_field_renderer_html_safe``
``ClearableFileInputTests.test_render_custom_template``

These are all obsoleted by the new rendering API.

``LiveWidgetTests``

This is more a ``ModelForm`` test than a ``Widget`` test and should be moved to a more appropriate place.

File: tests/forms_tests/widget_tests/test_dateinput.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+from django.utils import translation
+
+from .base import WidgetTest
+
+
+class DateInputTest(WidgetTest):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.widget = DateInput()
```
Comment (timgraham) [inline tests/forms_tests/widget_tests/test_dateinput.py]: I wonder if this initialization needs to be done in `setUpClass()` or if doing it at the class-level would work?
```
class DateInputTest(WidgetTest):
    widget = DateInput()
````

**Comment by timgraham** [comment]
Looks good. I just did a few cleanups along the way: http://dpaste.com/0DTFH6C
There are still a few lines > 119 chars if you could fix them. Hopefully we'll have [automated enforcement](https://code.djangoproject.com/ticket/23395) soon. Please merge at your convenience.

**Comment by prestontimmons** [comment]
Great. Thanks for reviewing this.

**Pull request merged**